### PR TITLE
Update BTDropInResult.fetchDropInResultForAuthorization

### PR DESCRIPTION
**Pull request by sestevens**
### Summary of changes

 - Updates to `BTDropIn.fetchDropInResultForAuthorization`:
   - Rename to `mostRecentPaymentMethod` (in Swift, this method was previously just `fetch`)
   - Change param from `authorization` to `clientToken` to clarify that a client token is required
   - Fix a bug where an empty `BTDropInResult` was returned if the customer had no vaulted payment methods. Now we return `nil`.
   - Return an error if the user passes an invalid client token. Previously, the completion block would never be called in this case.

 ### Checklist

 - [x] Added a changelog entry

- @sestevens 
- @scannillo

### CODE REVIEW

File: Sources/BraintreeDropIn/Models/BTDropInResult.m [hunk 1]
```diff
@@ -11,30 +11,67 @@
...
+static NSUserDefaults *_userDefaults = nil;
+
++ (NSUserDefaults *)userDefaults {
+    if (_userDefaults == nil) {
+        _userDefaults = [NSUserDefaults standardUserDefaults];
+    }
+    return _userDefaults;
+}
+
++ (void)setUserDefaults:(NSUserDefaults *)userDefaults {
```
Comment (sshropshire) [inline Sources/BraintreeDropIn/Models/BTDropInResult.m:29]: random: visible for testing?

**Review by sshropshire — APPROVED** [review]

File: Sources/BraintreeDropIn/Models/BTDropInResult.m (hunk 1 above)
Comment (sestevens) [reply Sources/BraintreeDropIn/Models/BTDropInResult.m:29]: Yep! That's specified for this property in `BTDropInResult_Internal.h`.

File: README.md [hunk 2]
```diff
@@ -145,25 +145,31 @@ let request =  BTDropInRequest()
...
-    })
+BTDropInResult.mostRecentPaymentMethod(forClientToken: authorization) { result, error in
+  guard let result = result, error == nil else {
+    // either an error occurred or the customer doesn't have any vaulted payment methods
+    return
+  }
+
+  if result.paymentOptionType == .applePay {
+    // Apple Pay is the most recently selected option
+    // Note that result.paymentMethod will be nil in this case; display Apple Pay button and tokenize using `BTApplePayClient`
```
Comment (scannillo) [inline README.md:163]: 👏  thank you for clarifying this for merchants

File: Sources/BraintreeDropIn/Public/BraintreeDropIn/BTDropInResult.h [hunk 3]
```diff
@@ -30,12 +44,16 @@ typedef void (^BTDropInResultFetchHandler)(BTDropInResult * _Nullable result, NS
...
-+ (void)fetchDropInResultForAuthorization:(NSString *)authorization handler:(BTDropInResultFetchHandler)handler;
+/**
+ * Fetch a `BTDropInResult` with the customer's most recently vaulted payment method.
+ * If the last payment method selected from Drop-in was Apple Pay, a `BTDropInResult` with
+ * `paymentOptionType == .applePay` will be returned in the completion block.
+ *
+ * @param clientToken Client token. Must be generated with a customer ID.
+ * @param completion The completion block, which passes back a result or an error. Both result and error may be nil if the customer does not have any vaulted payment methods.
+ */
++ (void)mostRecentPaymentMethodForClientToken:(NSString *)clientToken
```
Comment (scannillo) [inline Sources/BraintreeDropIn/Public/BraintreeDropIn/BTDropInResult.h:55]: Great idea renaming this param to `clientToken`

File: Sources/BraintreeDropIn/Models/BTDropInResult.m [hunk 4]
```diff
@@ -11,30 +11,67 @@
...
+    if (userDefaults != _userDefaults) {
+        _userDefaults = userDefaults;
+    }
+}
+
++ (void)mostRecentPaymentMethodForClientToken:(NSString *)clientToken
+                                   completion:(void (^)(BTDropInResult *result, NSError *error))completion {
+    BTAPIClient *apiClient = [[BTAPIClient alloc] initWithAuthorization:clientToken sendAnalyticsEvent:NO];
```
Comment (scannillo) [inline Sources/BraintreeDropIn/Models/BTDropInResult.m:37]: Idea - is it worth doing a client side validation here that the `clientToken` is actually a clientToken and returning back a more specific error?

Comment (sestevens) [reply Sources/BraintreeDropIn/Models/BTDropInResult.m:37]: That error will bubble up from the [base SDK](https://github.com/braintree/braintree_ios/blob/7ccda0f255230b8d427258766cb8041ba5ead5a4/Sources/BraintreeCore/BTAPIClient.m#L264).

**Review by scannillo — APPROVED** [review]

**Pull request merged**